### ISSUE TITLE
Votive Masses

### ISSUE THREAD

**Issue by tukusejssirs** [romcal/romcal#110]
@emagnier, I have no idea if I should add the celebration of Sts Cyril and Methodius to 14 February in `slovakia.js`.

The official liturgical calendar (called _Direktórium_) issue the by Liturgical Commission at the Conference of Slovak Bishops each year, says the following:
- 14 Feb: votive Mass of Sts Cyril, Monk, and Methodius, Bishop, Patrons of Europe;
- 5 July: Solemnity of Sts Cyril and Methodius, Slavic Missionaries.

The problem lays in the _votive Mass_ part. I know that _votive Mass_ is not a liturgical rank, but on the other hand it is still included in the official liturgical calendar.

Should I include it in #46? Or should I skip it until _some release after `v2.0.0`_, when we would deal with this by adding a special rank `VOTIVE_MASS`?

**Comment by jarosz** [comment]
https://en.wikipedia.org/wiki/General_Roman_Calendar#Slovakia - it states Feb 14 is a feast in Slovakia (as it's in Poland and I believe should be in all European countries).

**Comment by tukusejssirs** [comment]
@jarosz, I know what it states, but it is wrong. If you can understand a bit of Slovak, you can check out https://lc.kbs.sk/?den=20190214 and https://lc.kbs.sk/?den=20190705, which fully corresponds with the official Slovak Directory (liturgical calendar).

Anyway, I wanted to fix it there, but there is to much politics in the community about how it _should_ be done, so I was discouraged.

**Comment by emagnier** [comment]
Yes I saw that recently somewhere else for the Votive Mass of the Holy Spirit, and was wondering how this could be integrated.

Can we consider that `VOTIVE_MASS` could have the same rank/rules as an `OPT_MEMORY`?
I already played a bit with all these types in an experiment branch, but we need to rethink a bit first the way type dictionary (and all other constants) are managed in the code base (partially related to #109).

Anyway I'm in favor to include it soon. And depending on how we to the refactor, it will not break things or the API since it's just an addition. If we don't have time to include them in the v1.3.0, maybe this might be possible in v1.3.1.

**Comment by jarosz** [comment]
In feria per annum you can celebrate any votive mass, but not any saint memorial (only optional memorial and those saints, who are in Martyrology for that particular day), so it's not exactly the same.

In feria in Advent (to Dec 16), Christmastide, feria in Easter and memorials you can celebrate votive masses only if there's a real need of a group of faithful, but you can celebrate optional memorials and saints from Martyrology even if there's no need :)

**Comment by jarosz** [comment]
My educated guess is that Direktórium states u can celebrate a mass for Ciril and Methodius on Feb 14 in Slovakia even though the actual solemnity is said to be Jul 5 and it's called a votive mass not to break the first rule above.

**Comment by tukusejssirs** [comment]
> Can we consider that `VOTIVE_MASS` could have the same rank/rules as an `OPT_MEMORY`?

I am not a liturgist, but at least for our current needs we could do that (obviously, with the name `VOTIVE_MASS`).

@jarosz, I think so too and agree with you. I am not really for including the votive Masses in `romcal`, but I’d like to be as faithful to the official books as possible, and since the Direktorium states those three votive Masses explicitely, I’d rather have an option for that.

Anyway, I would like to suggest a new option that would enable outputing these votive Masses. By default, it would be turned off for every calendar without exception. Later, when we implement #20, we could have some ‘wrapper views’ that could some different settings.

**Comment by tukusejssirs** [comment]
Note to myself: when votive Masses are implemented, we might want to add the following key separately (currently we [mis-]use the celebration from `general`, which should be used for the votive Mass):
   - `slovakia.ts`: add a separate key for `Svätých Cyrila a Metoda, slovanských vierozvestov`.

**Comment by tukusejssirs** [comment]
Another reason to support votive Masses: in some particular calendars, we might celebrate a votive Masses:
- in honour of the Sacred Heart on the first Friday of a month;
- in honour of the Immaculate Heart of Mary on the first Saturday of a month.

See also the [_Rubrics for Privileged Votive Masses on First Thursdays & First Fridays_](https://acatholiclife.blogspot.com/2020/06/rubrics-for-privileged-votive-masses-on.html), albeit that article might deal with the rules in the Extra-Ordinary Form (`cal1962`), however, some rules still apply, like during Lent, neither of these two votive Masses are allowed.
